File: mobile/lib/models/plant.dart
class Plant {
  final int id;
  final String name;
  final String? species;
  final String? photo;
  final String? photoBase64;
  final int? ownerId;

  Plant({
    required this.id,
    required this.name,
    this.species,
    this.photo,
    this.photoBase64,
    this.ownerId,
  });

  factory Plant.fromJson(Map<String, dynamic> json) {
    return Plant(
      id: json['id'],
      name: json['name'] ?? '',
      species: json['species'],
      photo: json['photo'],
      photoBase64: json['photo_base64'],
      ownerId: json['owner_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'species': species,
      'photo': photo,
      'photo_base64': photoBase64,
      'owner_id': ownerId,
    };
  }
} 

File: api/models/plant.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text
from sqlalchemy.orm import relationship
from utils.database import Base


class Plant(Base):
    __tablename__ = "plants"
    id = Column(Integer, primary_key=True, autoincrement=True)
    # Champs anglais standard
    name = Column(String, nullable=False)
    species = Column(String, nullable=True)
    description = Column(String, nullable=True)
    photo_base64 = Column(Text, nullable=True)  # Image complète encodée en Base64
    photo_thumbnail = Column(Text, nullable=True)  # Thumbnail 150x150 pour listes
    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)

    # Relations
    owner = relationship("User", back_populates="owned_plants")
    photos = relationship("Photo", back_populates="plant", cascade="all, delete-orphan")

    # Relations avec les gardes
    cares = relationship("PlantCare", back_populates="plant")
